server/system/service/user.go

```diff
@@ -1035,3 +1051,119 @@ func toLabeledUsers(set []*types.User) []label.LabeledResource {
 
 	return ll
 }
+
+func (svc user) UploadAvatar(ctx context.Context, userID uint64, upload *multipart.FileHeader, initialColor ...string) (err error) {
+	var (
+		u       *types.User
+		att     *types.Attachment
+		uaProps = &userActionProps{user: &types.User{ID: userID}}
+	)
+
+	err = func() (err error) {
+		if u, err = loadUser(ctx, svc.store, userID); err != nil {
+			return
+		}
+
+		if u.Meta.AvatarID != 0 {
+			if err = svc.att.DeleteByID(ctx, u.Meta.AvatarID); err != nil {
+				return
+			}
+		}
+
+		if upload != nil {
+			file, err := upload.Open()
+			if err != nil {
+				return err
+			}
+
+			defer file.Close()
+
+			att, err = svc.att.CreateAuthAttachment(
+				ctx,
+				upload.Filename,
+				upload.Size,
+				file,
+				map[string]string{"key": types.AttachmentKindAvatar},
+			)
+
+			if err != nil {
+				return err
+			}
+		} else {
+			initial := processAvatarInitials(u)
```

Comment:
Make sure to handle sensible defaults for people with initials above char 127, like mr. Mesut Özil (for example).
What happens with him, does `CreateAvatarInitialsAttachment` throw an error?

Reply:
The new initials don't check whether the characters are of ASCII type. Instead, they check whether the character is a letter. If it is not a letter, the default initial is set to the letter 'C'.